src/test/java/net/snowflake/ingest/streaming/internal/ParquetValueParserTest.java

```diff
@@ -0,0 +1,552 @@
+
+    RowBufferStats rowBufferStats = new RowBufferStats();
+    ParquetValueParser.ParquetBufferValue pv =
+        ParquetValueParser.parseColumnValueToParquet(
+            new BigDecimal("12345.54321"),
+            testCol,
+            PrimitiveType.PrimitiveTypeName.DOUBLE,
+            rowBufferStats);
+
+    ParquetValueParserAssertionBuilder.newBuilder()
+        .parquetBufferValue(pv)
+        .rowBufferStats(rowBufferStats)
+        .expectedValueClass(Double.class)
+        .expectedSize(8.0f)
+        .expectedMinMax(Double.valueOf("12345.54321"))
+        .assertMatches();
+  }
+
+  @Test
+  public void parseValueDouble() {
+    ColumnMetadata testCol =
+        ColumnMetadataBuilder.newBuilder()
+            .logicalType("REAL")
+            .physicalType("DOUBLE")
+            .nullable(true)
+            .build();
+
+    RowBufferStats rowBufferStats = new RowBufferStats();
+    ParquetValueParser.ParquetBufferValue pv =
+        ParquetValueParser.parseColumnValueToParquet(
+            12345.54321d, testCol, PrimitiveType.PrimitiveTypeName.DOUBLE, rowBufferStats);
+
+    ParquetValueParserAssertionBuilder.newBuilder()
+        .parquetBufferValue(pv)
+        .rowBufferStats(rowBufferStats)
+        .expectedValueClass(Double.class)
+        .expectedSize(8.0f)
+        .expectedMinMax(Double.valueOf(12345.54321))
+        .assertMatches();
+  }
+
+  @Test
+  public void parseValueBoolean() {
+    ColumnMetadata testCol =
+        ColumnMetadataBuilder.newBuilder()
+            .logicalType("BOOLEAN")
+            .physicalType("SB1")
+            .nullable(true)
+            .build();
+
+    RowBufferStats rowBufferStats = new RowBufferStats();
+    ParquetValueParser.ParquetBufferValue pv =
+        ParquetValueParser.parseColumnValueToParquet(
+            true, testCol, PrimitiveType.PrimitiveTypeName.BOOLEAN, rowBufferStats);
+
+    ParquetValueParserAssertionBuilder.newBuilder()
+        .parquetBufferValue(pv)
+        .rowBufferStats(rowBufferStats)
+        .expectedValueClass(Boolean.class)
+        .expectedSize(1.0f)
+        .expectedMinMax(BigInteger.valueOf(1))
+        .assertMatches();
+  }
+
+  @Test
+  public void parseValueBinary() {
+    ColumnMetadata testCol =
+        ColumnMetadataBuilder.newBuilder()
+            .logicalType("BINARY")
+            .physicalType("LOB")
+            .nullable(true)
+            .build();
+
+    RowBufferStats rowBufferStats = new RowBufferStats();
+    ParquetValueParser.ParquetBufferValue pv =
+        ParquetValueParser.parseColumnValueToParquet(
+            "Length7".getBytes(), testCol, PrimitiveType.PrimitiveTypeName.BINARY, rowBufferStats);
+
+    ParquetValueParserAssertionBuilder.newBuilder()
+        .parquetBufferValue(pv)
+        .rowBufferStats(rowBufferStats)
+        .expectedValueClass(String.class)
+        .expectedSize(7.0f)
+        .expectedMinMax("Length7")
+        .assertMatches();
+  }
+
+  @Test
+  public void parseValueVariantToBinary() {
```

Comment:
Do you think this and next one can be deduplicated code-wise?

Reply:
yes, extracted code in `testJsonWithLogicalType(String logicalType)`